examples/ginkgo/ex1.cpp

```diff
@@ -194,24 +203,41 @@ int main(int argc, char *argv[])
       if (use_ginkgo_solver)
```

Comment:
It would be great if all of these nested conditionals could be refactored. It might always be a bit complicated since there are so many interacting options, but it's a bit hard to follow right now.

Reply:
I attemped to make the setup more clear with the `switch/case` setup in 2bd0031b8123582765d893b84fc04aaf58dd4260

Reply:
Nice, I think it's a good improvement.